src/backend/mpi/swdsm.cpp

```diff
@@ -1031,7 +1031,13 @@ void argo_initialize(unsigned long long size){
 		pthread_barrier_init(&threadbarrier[i],NULL,i);
 	}
 
-	cachesize = size+pagesize*CACHELINE;
+	cachesize = 0;
+	if(cache_size > argo_size) {
+		cachesize += argo_size;
+	} else {
+		cachesize += cache_size;
+	}
+	cachesize += pagesize*CACHELINE;
 	cachesize /= pagesize;
 	cachesize /= CACHELINE;
 	cachesize *= CACHELINE;
```

Comment:
For a future patch: There are two things going on here: First, rounding up to a whole number of cache blocks (and if the size already is, an extra cache block, which perhaps is not necessary). Second, division by `pagesize` to express `cachesize` in number of pages. These two actions should be separated for better clarity and the rounding performed by a function.

Reply:
See issue #24